src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx

```diff
@@ -0,0 +1,256 @@
+  const borderWidthRef = useRef(null);
+
+  const { bold, italic, underline, font, background, border } = merge(
+    {},
+    defaultProps.formValues,
+    formValues
+  );
+
+  return (
+    <div className={className}>
+      <div className={`${iotPrefix}--hotspot-text-style-tab__text-style`}>
+        <IconSwitch
+          onClick={() => onChange({ bold: !bold })}
+          data-testid="hotspot-bold"
+          selected={bold}
+          text={boldLabel}
+          renderIcon={TextBold}
+          index={0}
+          light={light}
+        />
+        <IconSwitch
+          name="italic"
+          onClick={() => onChange({ italic: !italic })}
+          data-testid="hotspot-italic"
+          selected={italic}
+          text={italicLabel}
+          renderIcon={TextItalic}
+          index={1}
+          light={light}
+        />
+        <IconSwitch
+          name="underline"
+          onClick={() => onChange({ underline: !underline })}
+          data-testid="hotspot-underline"
+          selected={underline}
+          text={underlineLabel}
+          renderIcon={TextUnderline}
+          index={2}
+          light={light}
+        />
+      </div>
+
+      <div className={`${iotPrefix}--hotspot-text-style-tab__row`}>
+        <ColorDropdown
+          id={`${iotPrefix}--hotspot-text-style-tab__font-color`}
+          titleText={fontLabel}
+          light={light}
+          selectedColor={font.color ?? fontColors?.[0]}
+          colors={fontColors}
+          onChange={(selected) => {
+            onChange({ font: selected });
+          }}
+        />
+
+        <NumberInput
+          id={`${iotPrefix}--hotspot-text-style-tab__font-size`}
+          min={0}
+          max={50}
+          value={font.size}
+          step={1}
+          light={light}
+          label={fontSizeLabel}
+          invalidText={fontSizeInvalid}
+          ref={fontSizeRef}
+          onChange={() => {
+            onChange({ font: { size: fontSizeRef.current.value } });
+          }}
+        />
+      </div>
+
+      <div className={`${iotPrefix}--hotspot-text-style-tab__row`}>
+        <ColorDropdown
+          id={`${iotPrefix}--hotspot-text-style-tab__background-color`}
+          titleText={backgroundLabel}
+          light={light}
+          selectedColor={background.color ?? backgroundColors?.[0]}
+          colors={backgroundColors}
+          onChange={(selected) => {
+            onChange({
+              background: selected,
+            });
+          }}
+        />
+
+        <NumberInput
+          id={`${iotPrefix}--hotspot-text-style-tab__background`}
+          label={fillOpacityLabel}
+          min={0}
+          max={100}
+          value={background.opacity}
+          ref={fillOpacityRef}
+          step={1}
+          light={light}
+          invalidText={fillOpacityInvalid}
+          onChange={() => {
+            onChange({
+              background: { opacity: fillOpacityRef.current.value },
```

Comment:
It seemed weird that we would have to use a ref to obtain the value, so I looked at the emitted event and it appears that carbon has added an `imaginaryTarget` element prop that contains the value. So you could do something like

```
onChange={(event) => {
  onChange({ background: event.imaginaryTarget.value})
}}
```